sdk/search/search-documents/src/searchIndexingBufferedSenderImpl.ts

```diff
@@ -354,7 +381,12 @@ class SearchIndexingBufferedSenderImpl<T> implements SearchIndexingBufferedSende
       // raise success event
       this.emitter.emit("batchSucceeded", result);
     } catch (e) {
-      if (this.isRetryAbleError(e) && retryAttempt < DEFAULT_RETRY_COUNT) {
+      if (this.isRetryAbleError(e) && retryAttempt <= this.maxRetries) {
+        const timeToWait =
+          retryAttempt * this.retryDelay <= this.maxRetryDelay
+            ? retryAttempt * this.retryDelay
+            : this.maxRetryDelay;
+        await delay(timeToWait);
```

Comment:
do we need to worry about adding jitter to prevent multiple clients from retrying in lockstep?

Reply:
I don't think so. The multiple clients should be independent.

Reply:
That's not the problem. The issue is that multiple independent clients can hammer a network resource and reduce availability by retrying simultaneously:

https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/

Jitter is helpful at spreading out load, especially when there are long delays at play.

Reply:
Thanks for explaining. I have added a random jitter to the code.